src/main/scala/com/ubirch/kafka/consumer/WithProcessRecords.scala

```diff
@@ -148,7 +148,7 @@ trait WithProcessRecords[K, V] {
       val error = failed.get()
       if (error.isDefined) {
         consumerRecords.partitions().asScala.foreach { p =>
-          val offset = Option(consumer.committed(p)).map(_.offset()).getOrElse(0L)
+          val offset = Option(consumer.committed(Set(p).asJava).get(p)).map(_.offset()).getOrElse(0L)
```

Comment:
can't write like this?
```scala
val partitionOffsets: java.util.Map[TopicPartition, OffsetAndMetadata]  = consumer.committed(consumerRecords.partitions())
partitionOffsets.asScala.foreach { case (p, offset) => 
    consumer.seek(p, offset)
}
```

Reply:
You are right, I was too focused on fixing just one line :)
I added a check for null to complete the tests. 

```
 val partitionOffsets: java.util.Map[TopicPartition, OffsetAndMetadata] = consumer.committed(consumerRecords.partitions())
        partitionOffsets.asScala.foreach {
          case (p, null) => consumer.seek(p, 0L)
          case (p, offset) => consumer.seek(p, offset)
        }
```